ably/error.go

```diff
@@ -0,0 +1,263 @@
+package ably
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/ably/ably-go/ably/proto"
+)
+
+func tostatus(code int) int {
```

Comment:
We use `statusCode` to represent the HTTP status code, so wouldn't it be `toStatusCode` ?